core/dbt/dataclass_schema.py

```diff
@@ -0,0 +1,927 @@
+import functools
```

Comment:
how much of this file do we want to keep around? It's just vendored hologram + some extras, right?

Reply:
Discussed this already, but for the record:  Unfortunately that hologram code is still used for constructing the jsonschemas that we use for validation. Would be nice to get rid of it or move it someplace. I pulled it into dbt for ease of programming and also because we will probably be changing it substantially in the not too distant future. It could be pulled back into hologram if we wanted to.

Reply:
I don't recall _exactly_ where we landed, but for my part, i'd be in favor of removing most of this code and reintroducing the hologram dependency. If we need to make changes in hologram, we can certainly do that!!

Reply:
I've removed the Hologram code and updated Hologram to work.